MapleServer2/Packets/PartyPacket.cs

```diff
@@ -0,0 +1,173 @@
+using Maple2Storage.Types;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Types;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MapleServer2.Packets
+{
+    public static class PartyPacket
+    {
+        public static Packet SendInvite(Player sender)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.PARTY)
+                .WriteByte(0xB) //Invite
+                .WriteUnicodeString(sender.Name)
+                .WriteShort(3843) //Unk
+                .WriteByte() //Unk
+                .WriteByte(); //Unk
+            return pWriter;
+        }
+
+        public static Packet Create(Player p)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.PARTY);
+
+            CreatePartyHeader(p, pWriter, 1);
+
+            CharacterListPacket.WriteCharacter(p, pWriter);
+            pWriter.WriteLong();
+            JobPacket.WriteSkills(pWriter, p);
+
+            return pWriter;
+        }
+
+        public static Packet CreateExisting(Player leader, List<Player> members)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.PARTY);
+
+            CreatePartyHeader(leader, pWriter, (byte)members.Count);
+
+            foreach (Player member in members)
+            {
+                CharacterListPacket.WriteCharacter(member, pWriter);
+                pWriter.WriteLong();
+                WriteSkills(pWriter, member);
+                if (member != members.Last())
+                {
+                    pWriter.WriteByte();
+                }
+            }
+            return pWriter;
+        }
+
+        //Generates the header code for Create
+        public static Packet CreatePartyHeader(Player p, PacketWriter pWriter, byte memberCount)
```

Comment:
can `Player p` be renamed `Player player`? The `p` can get confusing since sometimes `Party` is also used as `p`, same on the following packet methods